core/src/main/java/com/novoda/noplayer/internal/exoplayer/AdvertPlaybackState.java

```diff
@@ -42,9 +46,37 @@ static AdvertPlaybackState from(List<AdvertBreak> advertBreaks) {
         }
 
         adPlaybackState = adPlaybackState.withAdDurationsUs(advertBreaksWithAdvertDurations);
+        adPlaybackState = updateResumePositionInFirstGroup(adPlaybackState, advertBreakResumePositionMillis);
         return new AdvertPlaybackState(adPlaybackState, sortedAdvertBreaks);
     }
 
+    private static AdPlaybackState updateResumePositionInFirstGroup(AdPlaybackState state, long positionMillis) {
```

Comment:
Would it make sense to move this into `AdPlaybackState`, like `withAdDurationsUs()` so that we have `adPlaybackState.withResumePositionInFirstGroup()`?

Reply:
`AdPlaybackState` is an ExoPlayer class, that's why we have all this logic in a similar-sounding `AdvertPlaybackState` 😄